src/test_final_models_from_manual_folds_with_custom_loss.py

```diff
@@ -0,0 +1,303 @@
+    plot_name = plot_name.replace("/", "div")
+    plt.savefig(fname=os.path.join(plot_directory, plot_name))
+    plt.close()
+
+    # make a plot of TDD vs Y
+    x_tdd = np.arange(0, 501, 1)
+    plot_df = pd.DataFrame(columns=X_cols)
+    if "TDD" in str(X_cols):
+        plot_df["TDD"] = x_tdd
+        plt.plot(X_train["TDD"], y_train[y_lin_log.replace("log_", "")], ".", label="Train Data")
+        plt.plot(X_test["TDD"], y_test[y_lin_log.replace("log_", "")], ".", label="Test Data")
+        if "log_TDD" in str(X_cols):
+            plot_df["log_TDD"] = np.log(x_tdd + 1)
+        if "1/TDD" in str(X_cols):
+            plot_df["1/TDD"] = 1/(x_tdd + 1)
+
+        for x_cho in [0, 250, 500]:
+            for x_bmi in [12, 25, 45]:
+                line_name = f"BMI={x_bmi} CHO={x_cho}"
+                for x_col in X_cols:
+                    if ("TDD" not in x_col) & ("log_TDD" not in x_col):
+                        if "X_intercept" in x_col:
+                            plot_df["X_intercept"] = 1
+                        if "BMI" in str(X_cols):
+                            if "BMI" in x_col:
+                                plot_df["BMI"] = x_bmi
+                            if "log_BMI" in x_col:
+                                plot_df["log_BMI"] = np.log(x_bmi + 1)
+                        else:
+                            line_name = line_name.replace(f"BMI={x_bmi}", "BMI=NA")
+                        if "CHO" in str(X_cols):
+                            if "CHO" in x_col:
+                                plot_df["CHO"] = x_cho
+                            if "log_CHO" in x_col:
+                                plot_df["log_CHO"] = np.log(x_cho + 1)
+                        else:
+                            line_name = line_name.replace(f"CHO={x_cho}", "CHO=NA")
+
+                y_estimated = linear_regression_equation(fixed_parameters, plot_df[X_cols].values).astype("float64")
+                if "log" in y_lin_log:
+                    y_estimated = np.exp(y_estimated)
+                plt.plot(x_tdd, y_estimated, label=line_name)
+
+        plt.xlabel("TDD")
+        plt.ylabel(y_lin_log.replace("log_",""))
+        plt.title(
+            f"{equation_string}\n MDAPE={round(test_mdape, 3)}, MAPE={round(test_mape, 3)}, RMSE={round(test_rmse, 3)}"
+        )
+        plt.legend()
+        plot_name = f"TDD Plot MDAPE={round(test_mdape, 3)} {equation_string.replace(' ', '')}.png"
+        plot_name = plot_name.replace("/", "div")
+        plt.savefig(fname=os.path.join(plot_directory, plot_name))
+        plt.close()
+    else:
+        print("skipping this plot")
+
+
+# add in Rayhan Special for CIR
+# cir_rayhan = (0.39556 * CHO + 62.762) * TDD^(-0.71148)
+i = i + 1
+
+final_equations_df.loc[i, ["y", "X_intercept", "BMI", "CHO", "TDD", "beta_1/TDD"]] = [
+    "log_CIR_Rayhan",
+    "off",
+    "off",
+    "off",
+    "off",
+    "off",
+]
+
+if "CIR" in file_name:
+    # TODO: calculate the validation loss too
```

Comment:
Since we aren't using Rayhan's equation for the paper, are we able to resolve this TODO?